server.go

```diff
@@ -3616,6 +3629,44 @@ func (s *Server) applyCreateContinuousQueryCommand(m *messaging.Message) error {
 	return nil
 }
 
+// applyDropContinuousQueryCommand removes the continuous query from the database object and saves it to the metastore
+func (s *Server) applyDropContinuousQueryCommand(m *messaging.Message) error {
+	var c dropContinuousQueryCommand
+
+	mustUnmarshalJSON(m.Data, &c)
+
+	// retrieve the database and ensure that it exists
+	db := s.databases[c.Database]
+	if db == nil {
+		return ErrDatabaseNotFound
+	}
+
+	// loop through continuous queries and find the match
+	cqIndex := -1
+	for n, continuousQuery := range db.continuousQueries {
+		if continuousQuery.cq.Name == c.Name {
+			cqIndex = n
+			break
+		}
+	}
+
+	if cqIndex == -1 {
```

Comment:
So actually, this could happen, even if everything is OK. If two "drop CQ" commands for the same CQ hit the cluster at different nodes at the same time, one will win, but the other may still have gotten through before the first deletion finished. So if this happens, simply return nil here. In other words, make this operation idempotent.